examples/mugen/retrieval/eval.py

```diff
@@ -0,0 +1,120 @@
+        "bbox_smap_for_monsters": {"action": "store_true"},
+        "use_manual_annotation": {"action": "store_true"},
+        "use_auto_annotation": {"action": "store_true"},
+        "use_downsampled_trainset": {"action": "store_true"},
+        "fixed_start_idx": {"action": "store_true"},
+        "get_game_frame": {"action": "store_true"},
+        "get_seg_map": {"action": "store_true"},
+        "get_text_desc": {"action": "store_true"},
+        "debug": {"action": "store_true"},
+    },
+    "videoclip": {
+        "text_pretrained": {"action": "store_true"},
+        "text_trainable": {"action": "store_true"},
+        "text_model_name": {"type": str, "default": "distilbert-base-uncased"},
+        "text_model_config": {"type": json.loads, "default": None},
+        "text_padding_value": {"type": int, "default": 0},
+        "video_pretrained": {"action": "store_true"},
+        "video_trainable": {"action": "store_true"},
+        "video_pretrain_path": {"type": str, "default": PRETRAINED_S3D_KINETICS400_URL},
+        "proj_out_dim": {"type": int, "default": 256},
+        "proj_dropout": {"type": float, "default": 0.1},
+    },
+    "evaluation": {
+        "accelerator": {"type": str, "default": "auto"},
+        "devices": {"type": int, "default": 1},
+        "checkpoint_path": {
+            "type": str,
+            "default": "lightning_videoclip_mugen_ckpt.pt",
+        },
+    },
+}
+
+
+def parse_all_args():
+    parser = argparse.ArgumentParser()
+    for arguments_dict in arg_structure.values():
+        for arg_name, arg_options in arguments_dict.items():
+            parser.add_argument("--" + arg_name, **arg_options)
+    args = parser.parse_args()
+    return args
+
+
+def get_args_from_group(args, group_name):
+    """Utility for grouping arguments based on the structure in ``arg_structure``"""
+    return argparse.Namespace(
+        **{
+            arg_name: getattr(args, arg_name)
+            for arg_name in arg_structure[group_name].keys()
+        }
+    )
+
+
+def evaluate():
+    args = parse_all_args()
+    datamodule_args = get_args_from_group(args, "datamodule")
+    lightningmodule_args = get_args_from_group(args, "lightningmodule")
+    dataset_args = get_args_from_group(args, "dataset")
+    videoclip_args = get_args_from_group(args, "videoclip")
+    evaluation_args = get_args_from_group(args, "evaluation")
+
+    dataset_args = MUGENDatasetArgs(get_audio=False, **vars(dataset_args))
+    datamodule = MUGENDataModule(
+        dataset_args,
+        text_transform=BertTextTransform(),
+        video_transform=VideoTransform(),
+        **vars(datamodule_args),
+    )
+
+    model = VideoCLIPLightningModule(
+        **vars(lightningmodule_args), **vars(videoclip_args)
+    )
+    model = model.load_from_checkpoint(evaluation_args.checkpoint_path)
+
+    trainer = Trainer(
+        accelerator=evaluation_args.accelerator, devices=evaluation_args.devices
```

Comment:
Are these likely to be modified? (Put another way, have you tested the eval script on multiple devices?) If not you may not want to make this configurable

Reply:
did some testing + checked [lightning docs](https://pytorch-lightning.readthedocs.io/en/stable/common/lightning_module.html#testing:~:text=%3Dtest_dataloader)-,NOTE,single%20device%20to%20ensure%20each%20sample/batch%20gets%20evaluated%20exactly%20once.,-This%20is%20helpful) and will change to `accelerator="auto", devices=1`. i could also change accelerator to "cpu" but i think that will be an inconvenience when using the full test set